src/idpi/operators/curl.py

```diff
@@ -0,0 +1,122 @@
+import xarray as xr
+import numpy as np
+
+from .. import constants as const
+
+DEFAULT_DIFF_TYPE = "center"
+STENCILS = {
+    "left": {"m": slice(0, -2), "c": slice(1, -1), "p": slice(1, -1)},
+    "center": {"m": slice(0, -2), "c": slice(1, -1), "p": slice(2, None)},
+    "right": {"m": slice(1, -1), "c": slice(1, -1), "p": slice(2, None)},
+}
```

Comment:
I modified the handling of the stencils to be more static and less dependent on the mutation of the global state. That can be a pain to work with in a concurrent execution context.